asf_public_discourse_home_decarbonisation/analysis/buildhub/extract_faqs.py

```diff
@@ -0,0 +1,245 @@
+    <forum>: The forum that we are looking at (e.g. bh or MSE).
+    <category>: The category within the forum (e.g. energy).
+    <collection_date_time>: The date time format, this differs between 'YYYY_MM_DD' for MSE and 'YYMMDD' for bh.
+Example Usage:
+    For MSE:
+        'python extract_faqs.py --forum mse --category energy --collection_date_time 2023_11_15'
+    For buildhub:
+        'python extract_faqs.py --forum bh --category combined_data --collection_date_time 231120'
+    Note: In these run commands note the different format in date-time format for the different forums.
+"""
+import pandas as pd
+import re
+import nltk
+from nltk.tokenize import sent_tokenize
+from typing import List, Tuple
+import argparse
+from asf_public_discourse_home_decarbonisation.getters.bh_getters import (
+    get_bh_category_data,
+)
+from asf_public_discourse_home_decarbonisation.getters.mse_getters import (
+    get_mse_category_data,
+)
+import os
+from asf_public_discourse_home_decarbonisation import PROJECT_DIR
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def create_argparser() -> argparse.ArgumentParser:
+    """
+    Creates an argument parser that can receive the following arguments:
+    - category: category or sub-forum (defaults to "119_air_source_heat_pumps_ashp")
+    - collection_date_time: collection date/time (defaults to "231120")
+    Returns:
+        argparse.ArgumentParser: argument parser
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--forum",
+        help="Forum that we are looking at",
+        default="bh",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--category",
+        help="Category or sub-forum",
+        default="119_air_source_heat_pumps_ashp",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--collection_date_time",
+        help="Collection date/time",
+        default="231120",
+        type=str,
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def extract_questions(
+    sentences: List[str], num_of_words: int = 5, min_length: int = 5
+) -> List[str]:
+    """
+    Extracts and returns a curated list of questions from a list of sentences, specifically filtering out certain types of questions. This function is designed to identify and include questions that are more substantial and exclude shorter, less informative questions such as "any thoughts?" or "why?".
+
+    The function utilises a combination of punctuation analysis (identifying sentences ending with '?') and the presence of certain question words at the beginning. However, it also employs a set of criteria to exclude questions that are deemed too brief or lacking in content, even if they technically qualify as questions.
+
+    Args:
+        sentences (List[str]): A list of sentences from which questions are to be extracted.
+        num_of_words (int, optional): The maximum number of words allowed for a question to be considered valid. Defaults to 5.
```

Comment:
should be minimum no? we're only keeping questions with a minimum number of X words
```suggestion
        num_of_words (int, optional): The minimum number of words allowed for a question to be considered valid. Defaults to 5.
```